### PR TITLE
ini_file: Don't report changed=true when removing if nothing is changed.

### LINKED ISSUE #4154: ini_file: reports change when told to remove non-existing section
### Summary

When using ini_file to remove a section from an ini file, changes are reported even if no changes were done/required.

### Issue Type

Bug Report

### Component Name

ini_file

### Ansible Version

```console (paste below)
$ ansible --version
ansible [core 2.11.6] 
  config file = /home/user/ansible/ansible.cfg
  configured module search path = ['/home/user/ansible/library']
  ansible python module location = /usr/lib/python3/dist-packages/ansible
  ansible collection location = /home/user/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.8.10 (default, Nov 26 2021, 20:14:08) [GCC 9.3.0]
  jinja version = 2.10.1
  libyaml = True
```

### Community.general Version

```console (paste below)
$ ansible-galaxy collection list community.general
# /usr/lib/python3/dist-packages/ansible_collections
Collection        Version
----------------- -------
community.general 3.7.0
```

### Configuration

```console (paste below)
$ ansible-config dump --only-changed
ANSIBLE_NOCOWS(/home/user/ansible/ansible.cfg) = True
CALLBACKS_ENABLED(env: ANSIBLE_CALLBACKS_ENABLED) = ['profile_tasks']
DEFAULT_FORKS(/home/user/ansible/ansible.cfg) = 150
DEFAULT_HOST_LIST(/home/user/ansible/ansible.cfg) = ['/home/user/ansible/inventory-production.ini']
DEFAULT_LOCAL_TMP(/home/user/ansible/ansible.cfg) = /home/user/.ansible/tmp/ansible-local-19356977961lflx
DEFAULT_MODULE_PATH(/home/user/ansible/ansible.cfg) = ['/home/user/ansible/library']
DEFAULT_MODULE_UTILS_PATH(/home/user/ansible/ansible.cfg) = ['/home/user/ansible/module_utils']
DEFAULT_POLL_INTERVAL(/home/user/ansible/ansible.cfg) = 15
DEFAULT_REMOTE_USER(/home/user/ansible/ansible.cfg) = ansible
DEFAULT_ROLES_PATH(/home/user/ansible/ansible.cfg) = ['/home/user/ansible/roles']
DEFAULT_STDOUT_CALLBACK(env: ANSIBLE_STDOUT_CALLBACK) = yaml
DEFAULT_TIMEOUT(/home/user/ansible/ansible.cfg) = 45
DEFAULT_VAULT_IDENTITY_LIST(/home/user/ansible/ansible.cfg) = ['prod@vault-prod', 'dev@vault-dev']
DIFF_ALWAYS(env: ANSIBLE_DIFF_ALWAYS) = True
GALAXY_ROLE_SKELETON(/home/user/ansible/ansible.cfg) = /home/user/ansible/skeleton-role
HOST_KEY_CHECKING(/home/user/ansible/ansible.cfg) = False
INVENTORY_ANY_UNPARSED_IS_FAILED(env: ANSIBLE_INVENTORY_ANY_UNPARSED_IS_FAILED) = True
INVENTORY_UNPARSED_IS_FAILED(env: ANSIBLE_INVENTORY_UNPARSED_FAILED) = True
RETRY_FILES_SAVE_PATH(/home/user/ansible/ansible.cfg) = /home/user/ansible/.ansible-retry
```

### OS / Environment

Ubuntu 20.04

### Steps to Reproduce

```yaml (paste below)
- name: reset test file
  ansible.builtin.copy:
      content: |
        [section1]
        foo=bar

        [section2]
        bar=foo

        [section3]
        haha=hoho
      dest: /tmp/foo.ini

- name: remove section
  community.general.ini_file:
    path: /tmp/foo.ini
    section: "section2"
    state: absent

- name: remove section again
  community.general.ini_file:
    path: /tmp/foo.ini
    section: "section2"
    state: absent

- name: remove section again w. loop
  community.general.ini_file:
    path: /tmp/foo.ini
    section: "{{ item.section }}"
    state: absent
  with_items:
    - { section: "section2" }
```

### Expected Results

I would expect task `remove section again` and task `remove section again w. loop` to report no changes

### Actual Results

Task `remove section again` and task `remove section again w. loop` reports changes

```console (paste below)
TASK [reset test file] ***********************************************
fredag 04 februar 2022  14:17:20 +0100 (0:00:02.062)       0:00:02.107
--- before: /tmp/foo.ini
+++ after: /home/user/.ansible/tmp/ansible-local-1936585ojv905_r/tmpftrnqd9q
@@ -1,5 +1,8 @@
 [section1]
 foo=bar
  
+[section2]
+bar=foo
+
 [section3]
 haha=hoho

changed: [node01.example.com]

TASK [remove section] ************************************************
fredag 04 februar 2022  14:17:22 +0100 (0:00:01.235)       0:00:03.343
--- before: /tmp/foo.ini (content)
+++ after: /tmp/foo.ini (content)
@@ -1,8 +1,5 @@
 [section1]
 foo=bar
  
-[section2]
-bar=foo
-
 [section3]
 haha=hoho

changed: [node01.example.com]

TASK [remove section again] ******************************************
fredag 04 februar 2022  14:17:22 +0100 (0:00:00.355)       0:00:03.699
changed: [node01.example.com]

TASK [remove section again w. loop] **********************************
fredag 04 februar 2022  14:17:22 +0100 (0:00:00.259)       0:00:03.958
changed: [node01.example.com] => (item={'section': 'section2'})


```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by jameslivulpi**
##### SUMMARY
fixes #4154 

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

ini_file

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: changelogs/fragments/4154-ini_file_changed.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - ini_file - When removing nothing don't report changed.
```
Comment (felixfontein) [inline changelogs/fragments/4154-ini_file_changed.yml:2]:
```suggestion
  - ini_file - when removing nothing do not report changed (https://github.com/ansible-collections/community.general/issues/4154).
```

File: plugins/modules/files/ini_file.py [hunk 2]
```diff
@@ -367,9 +367,10 @@ def do_ini(module, filename, section=None, option=None, values=None,
                     section_lines = new_section_lines
         else:
             # drop the entire section
-            section_lines = []
-            msg = 'section removed'
-            changed = True
+            if len(section_lines) != 0:
```
Comment (felixfontein) [inline plugins/modules/files/ini_file.py:370]: More Pythonic would be:
```suggestion
            if section_lines:
```

**Review by felixfontein — COMMENTED** [review]
It's probably a good idea to add a regression test for this. The module has its integration tests in tests/integration/targets/ini_file/. (Sounds like a good idea to have `state: absent` idempotency tests - re-run a `state: absent` task and check that the second iteration has `changed: false`.)

**Comment by jameslivulpi** [comment]
Working on the integration test. Having some strange env issue but will get it asap.

**Comment by jameslivulpi** [comment]
Sorry for delay on this. Added absent idempotency test.

**Review by felixfontein — APPROVED** [review]
Looks good to me! If nobody complains, I'll merge in a couple of days.

**Pull request merged**